From 1ad4230fe35863211dd2a8d16db5072bf633ae21 Mon Sep 17 00:00:00 2001
From: "raviprabhat.tcs" <developer1-tc@uidai.net.in>
Date: Tue, 31 Mar 2026 16:24:58 +0530
Subject: [PATCH] 2. Creating Dummy Test Double Using Mockito

---
 .../mockito/junit/dummy/DummyTest.java        | 39 +++++++++++++++++++
 1 file changed, 39 insertions(+)

diff --git a/src/test/java/com/hubberspot/mockito/junit/dummy/DummyTest.java b/src/test/java/com/hubberspot/mockito/junit/dummy/DummyTest.java
index c5fabe0..f8cc0b1 100644
--- a/src/test/java/com/hubberspot/mockito/junit/dummy/DummyTest.java
+++ b/src/test/java/com/hubberspot/mockito/junit/dummy/DummyTest.java
@@ -3,8 +3,11 @@ package com.hubberspot.mockito.junit.dummy;
 
 
 import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
 
 import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Collection;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
@@ -32,4 +35,40 @@ public class DummyTest {
 //
         assertEquals(1, bookService.findNumberOfBooks());
     }
+
+
+    @Test
+    public void dummyTestDoubleWithMockito(){
+
+
+//      here we are creating BookRepository mock with the help of Mockito,
+//      or simply we are mocking BookRepository
+        BookRepository bookRepository = Mockito.mock(BookRepository.class);
+
+//      Mocking EmailService
+        EmailService emailService = Mockito.mock(EmailService.class);
+//      To create bookService object -> we pass mock bookRepository and mock emailService-> Hence we bypass
+//      the need of actual bookRepository and emailService dependency
+        BookService bookService = new BookService(bookRepository, emailService);
+
+
+        Book book1 = new Book("1234", "Mockito In Action", 250, LocalDate.now());
+        Book book2 = new Book("51237", "JUnit5 In Action", 200, LocalDate.now());
+
+        Collection<Book> books = new ArrayList<>();
+        books.add(book1);
+        books.add(book2);
+
+//      When you write Mockito.when(bookRepository.findAll()).thenReturn(books);,
+//      you're defining the behavior of the mock (i.e., the fake implementation of BookRepository).
+//      When the production code under test (bookService.findNumberOfBooks()) calls bookRepository.findAll(), Mockito
+//      intercepts that call and returns the value you defined (books).
+        Mockito.when(bookRepository.findAll()).thenReturn(books);
+
+//      Goal is to test findNumberOfBooks() method defined in BookService class
+//      so here we are calling findNumberOfBooks() method
+        assertEquals(2, bookService.findNumberOfBooks());
+
+
+    }
 }
-- 
2.41.0.windows.1

