From be0482a54ae894ce272ce5dd908a71a91ed27758 Mon Sep 17 00:00:00 2001
From: "raviprabhat.tcs" <developer1-tc@uidai.net.in>
Date: Tue, 31 Mar 2026 17:51:45 +0530
Subject: [PATCH] 3. Creating Stub Test Double using Mockito

---
 .../mockito/junit/stub/StubTest.java          | 43 +++++++++++++++++++
 1 file changed, 43 insertions(+)

diff --git a/src/test/java/com/hubberspot/mockito/junit/stub/StubTest.java b/src/test/java/com/hubberspot/mockito/junit/stub/StubTest.java
index c9c605d..fe6cb9f 100644
--- a/src/test/java/com/hubberspot/mockito/junit/stub/StubTest.java
+++ b/src/test/java/com/hubberspot/mockito/junit/stub/StubTest.java
@@ -1,8 +1,12 @@
 package com.hubberspot.mockito.junit.stub;
 
 import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
 
+import java.util.ArrayList;
+import java.util.Collection;
 import java.util.List;
+import java.time.LocalDate;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
@@ -24,5 +28,44 @@ public class StubTest {
 
 
 
+    }
+
+    @Test
+    public void testDoubleStubWithMockito(){
+
+//      Mocking bookRepository
+        BookRepository bookRepository = Mockito.mock(BookRepository.class);
+        BookService bookService = new BookService(bookRepository);
+
+
+        Book book1 = new Book("1234", "Mockito In Action", 500, LocalDate.now());
+        Book book2 = new Book("5127", "Junit5 In Action", 400, LocalDate.now());
+
+        List<Book> newBooks = new ArrayList<>();
+        newBooks.add(book1);
+        newBooks.add(book2);
+
+//      Defining behaviour for mock object -> bookRepository
+//
+        Mockito.when(bookRepository.findNewBooks(7)).thenReturn(newBooks);
+
+//        here we are calling BookService method -> getNewBooksWithAppliedDiscount() and using assertEquals() to verify business logic
+//        written in BookService class
+
+//        when getNewBooksWithAppliedDiscount() method which is defined in BookService is called
+//        then mock bookRepository comes into picture -> And we already defined the behaviour of mock bookRepository
+
+//        Inside getNewBooksWithAppliedDiscount() method -> we have written a line
+//        List<Book> newBooks = bookRepository.findNewBooks(days);
+//        so here mock bookRepository will be called and mock bookRepository will call findNewBooks() method
+//        and it will return a list object newBooks, and we can check discount logic using assert() method
+        List<Book> newBooksWithAppliedDiscount = bookService.getNewBooksWithAppliedDiscount(10, 7);
+
+        assertEquals(2, newBooksWithAppliedDiscount.size());
+        assertEquals(450, newBooksWithAppliedDiscount.get(0).getPrice());
+        assertEquals(360, newBooksWithAppliedDiscount.get(1).getPrice());
+
+
+
     }
 }
-- 
2.41.0.windows.1

